ext/web/08_text_encoding.js

```diff
@@ -301,6 +301,14 @@ class TextDecoderStream {
           return PromiseReject(err);
         }
       },
+      cancel: (_reason) => {
+        try {
+          const _ = this.#decoder.decode();
+          return PromiseResolve();
+        } catch (err) {
+          return PromiseReject(err);
+        }
+      },
```

Comment:
I believe something like this can be used to simplify the construction:

```suggestion
      cancel: async (_reason) => {
        const _ = this.#decoder.decode();
      },
```

Reply:
I see that this entire class is using the other style -- I'm not sure why so let's leave as-is. This is probably a good follow-up issue. I'll merge it!

Reply:
Yes, I tried to follow the existing style. Not sure why it is preferred here, though.